Sources/SwiftProtobuf/BinaryEncoder.swift

```diff
@@ -125,12 +127,24 @@ internal struct BinaryEncoder {
 
     // Write a string field, including the leading index/tag value.
     mutating func putStringValue(value: String) {
-        let count = value.utf8.count
-        putVarInt(value: count)
-        for b in value.utf8 {
-            pointer.storeBytes(of: b, as: UInt8.self)
-            pointer = pointer.advanced(by: 1)
-        }
+        #if swift(>=5.0)
+            // If the String does not support an internal representation in a form
+            // of contiguous storage, body is not called and nil is returned.
+            let isAvailable = value.utf8.withContiguousStorageIfAvailable { (body: UnsafeBufferPointer<UInt8>) -> Int in
+                putVarInt(value: body.count)
+                return append(contentsOf: UnsafeRawBufferPointer(body))
+            }
+        #else
+            let isAvailable: String? = nil
```

Comment:
Minor: This should be `Int?` to match the type of the `withContiguousStorage...` call above.

Reply:
That's on me.  I suggested reflowing the conditional and didn't full think about what the type was, so I just went with `String?`, the reality is it doesn't matter since it just needs to define it as some thing that sorta aligns with the other call from the POV of being an optional that can be `if`ed on.